src/lightning_app/components/database/server.py

```diff
@@ -172,6 +180,15 @@ def db_url(self) -> Optional[str]:
             return f"http://{self.internal_ip}:{self.port}"
         return self.internal_ip
 
-    def on_exit(self):
-        self.drive.put(self.db_filename)
+    def periodic_store_database(self, store_interval):
+        while True:
+            self.store_database()
+            time.sleep(store_interval)
+
+    def store_database(self):
+        with self._store_lock:
```

Comment:
Is the lock here to prevent exit() and the _store_thread to interact in a race condition?

Reply:
Yes correct